### PR TITLE
CRM/Contribute - Cleanup boolean expressions

### PR DESCRIPTION
**Pull request by colemanw**
Broken out from #16814.

### CODE REVIEW

File: CRM/Contribute/Form/Contribution.php [hunk 1]
```diff
@@ -1683,7 +1683,7 @@ protected function submit($submittedValues, $action, $pledgePaymentID) {
       $action,
       $pledgePaymentID,
       $contribution->id,
-      (CRM_Utils_Array::value('option_type', $formValues) == 2) ? TRUE : FALSE,
+      ($formValues['option_type'] ?? 0) == 2,
```
Comment (seamuslee001) [inline CRM/Contribute/Form/Contribution.php]: This is the only one I'm not 100% sure about @eileenmcnaughton thoughts?

Comment (colemanw) [reply CRM/Contribute/Form/Contribution.php]: It's the same logic in both expressions.

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/Contribution.php]: Yes - it first resolved $formValues['option_type'] to it's value OR 0 and then compares with 2

Comment (seamuslee001) [reply CRM/Contribute/Form/Contribution.php]: Guess it just seemed odd to me resolving to 0 when CRM_Utils_Array::value would have resolved to NULL but i guess for this purpose they are interchangeable values

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/Contribution.php]: Yep - probably NULL would have been more readable

**Pull request merged**